install_config/persistent_storage/persistent_storage_nfs.adoc

```diff
@@ -11,124 +11,244 @@
+implementing NFS volumes.
+
+Developers request NFS storage by referencing, in the `volumes` section of their
+pod spec, either a _PersistentVolumeClaim_ by name, or the NFS volume plug-in directly.
+
+The *_/etc/exports_* file, on the NFS server, contains the accessible NFS directories.
+The target NFS directory has the familiar POSIX owner and group ids. The Openshift NFS
+plug-in mounts the container's NFS directory with the same POSIX ownership and permissions
+found on the exported NFS directory. However, the container is not run with its effective
+UID equal to the owner of the NFS mount, which is the desired behavior.
+
+As an example, if the target NFS directory appears as:
+[[nfs-export]]
+```
+#on the nfs server:
+# ls -lZ /opt/nfs -d
+drwxrws---. nobody 5555 unconfined_u:object_r:usr_t:s0   /opt/nfs
+
+# id nobody
+uid=99(nobody) gid=99(nobody) groups=99(nobody)
+```
+then the container will need to match selinux labels, and either run with a UID of 99
+(_nobody_ owner), or with 5555 in its supplemental groups, in order to access the directory.
+
+*Note:* the owner id of 99 is used as an example. Even though NFS's _root_squash_ maps *_root_* (0) to *_nobody_* (99), NFS exports can have arbitrary owner ids.
+ Owner 99 is not required for NFS exports.
+
+=== User IDs
+User ids can be defined in the container image or in the pod spec. The
+link:pod_security_context.html#user-id[pod and storage security] document covers
+controlling storage access based on user ids.
+
+In the above NFS export, the container needs it's UID set to 99 (ignoring group ids for the moment), so the
+following fragement can be added to the pod spec:
+```
+spec:
+  containers: <1>
+  - name: ...
+    securityContext:
+      runAsUser: 99  #nobody
+```
+<1> pods contain a `securtityContext` specific to each container (shown above), and a global
+`securityContext` which applies to all containers defined in the pod.
+
+Assuming the _default_ project and the _restricted_ SCC, the pod's requested user id of 99
+will, unfortunetely, *not* be allowed, and therefore the pod will fail. The pod fails because:
+
+- it requests 99 as its user id,
+- all SCCs available to the pod are examined to see which SCC will allow a user id of 99
+(actually, all policies of the SCCs are checked but the  focus here is on user id),
+- since all available SCCs use _MustRunAsRange_ for their `runAsUser` strategy, uid
+range checking is required, 
+- 99 is not included in the SCC or namespace's user id range, so the pod fails.
+
+It's generally considered a good practice to *not* modify the predefined SCCs. The preferred 
+way to fix this situation is to create a custom SCC, as described in 
+link:pod_security_context#scc-runasuser[pod security and custom SCCs].
+A custom SCC can be created such that a min and max user ids are defined, UID range checking is still enforced, and the UID of 99 will be allowed.
```

Comment:
The anyuid scc has a priority set so it will always be tested first IF the actor creating the pod has access to it via user or group grants - with the assumption that the pod is not making other privileged requests that anyuid would  deny.  By default it is only granted to admins but it is our recommended way of running dockerhub images that require root (https://docs.openshift.org/latest/admin_guide/manage_scc.html#enable-dockerhub-images-that-require-root)